test/config-luatest/credentials_applier_test.lua

```diff
@@ -464,26 +462,28 @@ g.test_sync_privileges = function(g)
         },
     })
     local dir = treegen.prepare_directory(g, {}, {})
-    child:roundtrip(("box.cfg{work_dir = %q}"):format(dir))
+    child:assert_eval(("box.cfg{work_dir = %q}"):format(dir), nil)
 
     local name = "myuser"
-    child:roundtrip(("box.schema.user.create(%q)"):format(name))
+    child:assert_eval(("box.schema.user.create(%q)"):format(name), nil)
     for _, command in ipairs(box_configuration) do
         local action, perm, obj_type, obj_name = unpack(command)
         local opts = "{if_not_exists = true}"
-        child:roundtrip(("box.schema.user.%s(%q, %q, %q, %q, %s)"):format(
-                         action, name, perm, obj_type, obj_name, opts))
+        child:assert_eval(("box.schema.user.%s(%q, %q, %q, %q, %s)"):format(
+                           action, name, perm, obj_type, obj_name, opts), nil)
```

Comment:
Yes, it does seem more natural to have `:eval()` raise on error inside (but on the other hand, one could argue that `it` is about testing console with the exact output and having error being returned in the same way as everything else is OK). Anyway, let's have this somewhat complicated logic of error detection hidden inside helper, it would be safer.

Reply:
I added error rethrow, but decided to remove patch that renames `roundtrip()` to `assert_eval()` as the main reason behind it (need to check for `nil`) vanished.

Reply:
It just comes into my head that we should raise `reply[1].error`, not just `reply[1]`. Sorry for misguiding.

Reply:
> It just comes into my head that we should raise reply[1].error, not just reply[1]. Sorry for misguiding.

Oh, right. I missed that too, just checked that it errored when expected.

Reply:
> (Please, also update the pull request title/description.)

Fixed.